doc/psud/PSU_daemon_design.md

```diff
@@ -37,40 +49,99 @@ PSU daemon data collection flow diagram:
 
 Now psud collects PSU data via platform API, and it also support platform plugin for backward compatible. All PSU data will be saved to redis database for further usage.
 
+### 2.1 PSU data collection specific to PSU power exceeding check
+
+We will leverage the existing framework of PSU daemon, adding corresponding logic to perform PSU power check.
+
+Currently, PSU daemon is waken up periodically, executing the following flows (flows in bold are newly introduced by the feature):
+
+1. Check the PSUs' physical entity information and update them into database
+2. Check the PSUs' present and power good information and update them to database
+   - __It will check the capability of PSU power check via reading PSU power thresholds when a new PSU is detected.__
+3. Check and update the PSUs' data
+   - Fetch voltage, current, power via calling platform API
+   - __Perform PSU power checking logic__
+   - Update all the information to database
+
+We will detail the new flows in the following sections.
+
+#### New PSU is detected
+
+Basically, there are two scenarios in which a new PSU can be detected:
+
+- On PSU daemon starting, all PSUs installed on the switch are detected
+- On new PSU pulgged, the new PSU is detected
+
+When one or more new PSUs is detected and power is good, PSU daemon tries retrieving the warning and critical threshold for each PSU installed on the switch.
+
+The PSU power checking will not be checked for a PSU if `NotImplemented` exception is thrown or `None` is returned while either threshold is being retrieved
+
+#### Alarm raising and clearing threshold
+
+We use asymmetric thresholds between raising and clearing the alarm for the purpose of creating a hysteresis and avoiding alarm flapping.
+
+- an alarm will be raised when a PSU's power is rising accross the critical threshold
+- an alarm will be cleared when a PSU's power is dropping across the warning threshold
```

Comment:
Thank you for the HLD presentation.  Based on today's discussion, I would like to propose renaming the "warning threshold" to something that may be more clear based on the implementation (e.g. "nominal_power_threshold") and update the document accordingly.

Reply:
Hi Jeff
Thank you for your comment. After internal discussion, we decide to keep the current name. The reason is that it is not completely normal but indicates a risk once the PSU power consumption raises across the warning threshold even though no alarm information is raised in this situation.

Reply:
@stephenxs can we rename as "warning-suppress" level? i am ok with @jeffhtgt suggestion for 'nominal_power_threshold'

Reply:
I do not suggest using `nominal_power_threshold` since it is still represent a warning situation even though we do not raise alarm now.

Reply:
let's use `power_warning_suppress_threshold', ok? @prgeor